src/ray/raylet/task_spec.h

```diff
@@ -109,6 +109,14 @@ class TaskSpecification {
                     int64_t num_returns,
                     const std::unordered_map<std::string, double> &required_resources);
 
+  /// Create a task specification from a pointer.
+  ///
+  /// \param spec The specification data from a pointer.
+  /// \param spec_size The specification data size.
+  TaskSpecification(const uint8_t *spec, size_t spec_size) {
```

Comment:
Actually, we're doing the same thing in #2526 as well. We should make this

```cpp
TaskSpecification(const std::string &string);
```

Reply:
I think it is not conflict.
If we only use `TaskSpecification(const std::string &string);`
we must pack the pointer into a string [here](https://github.com/ant-tech-alliance/ray/blob/java_raylet_2/src/local_scheduler/lib/java/org_ray_spi_impl_DefaultLocalSchedulerClient.cc#L93), and call  string-verson ctor. 
but in string-version ctor, we should unpack it to data-pointer.
I think we should remaining the 2 version ctor, string-version and data-pointer version.
How do you like it? Or do u have a better idea? @robertnishihara

Reply:
It' ok. :)